nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java

```diff
@@ -81,9 +84,8 @@ public class OptionalEmptinessHandler extends BaseNoOpHandler {
   @Override
   public boolean onOverrideMayBeNullExpr(
       NullAway analysis, ExpressionTree expr, VisitorState state, boolean exprMayBeNull) {
-    if (expr.getKind() == Tree.Kind.METHOD_INVOCATION
-        && optionalIsGetCall((Symbol.MethodSymbol) ASTHelpers.getSymbol(expr), state.getTypes())) {
-      return true;
+    if (!exprMayBeNull && expr.getKind() == Tree.Kind.METHOD_INVOCATION) {
+      return optionalIsGetCall((Symbol.MethodSymbol) ASTHelpers.getSymbol(expr), state.getTypes());
```

Comment:
small refactoring i thought may make sense to safe a few cycles, but happy to revert this part if not worth it (since admittedly hard to review)

Reply:
removed since it may be controversial